File: src/main/java/um/haberes/report/model/haberes/core/PersonaDto.java
package um.haberes.report.model.haberes.core;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.text.MessageFormat;
import java.time.OffsetDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PersonaDto {

    private Long legajoId;

    private BigDecimal documento = BigDecimal.ZERO;

    private String apellido = "";

    private String nombre = "";

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime nacimiento;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime altaDocente;

    private int ajusteDocente = 0;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime altaAdministrativa;

    private int ajusteAdministrativo = 0;

    private String estadoCivil = "";

    private Integer situacionId;

    private byte reemplazoDesarraigo = 0;

    private byte mitadDesarraigo = 0;

    private String cuil = "";

    private int posgrado = 0;

    private int estado = 0;

    private String liquida = "";

    private int estadoAfip = 0;

    private Integer dependenciaId;

    private String salida;

    private Long obraSocial;

    private Integer actividadAfip;

    private Integer localidadAfip;

    private int situacionAfip = 0;

    private Integer modeloContratacionAfip;

    private byte directivoEtec = 0;

    private DependenciaDto dependencia;

    private AfipSituacionDto afipSituacion;

    public String getApellidoNombre() {
        return MessageFormat.format("{0}, {1}", this.apellido, this.nombre);
    }

}

File: src/main/java/um/haberes/report/kotlin/dto/PersonaDto.kt
package um.haberes.report.kotlin.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigDecimal
import java.text.MessageFormat
import java.time.OffsetDateTime

data class PersonaDto(

    var legajoId: Long? = null,
    var documento: BigDecimal = BigDecimal.ZERO,
    var apellido: String = "",
    var nombre: String = "",
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var nacimiento: OffsetDateTime? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var altaDocente: OffsetDateTime? = null,
    var ajusteDocente: Int = 0,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var altaAdministrativa: OffsetDateTime? = null,
    var ajusteAdministrativo: Int = 0,
    var estadoCivil: String = "",
    var situacionId: Int? = null,
    var reemplazoDesarraigo: Byte = 0,
    var mitadDesarraigo: Byte = 0,
    var cuil: String = "",
    var posgrado: Int = 0,
    var estado: Int = 0,
    var liquida: String = "",
    var estadoAfip: Int = 0,
    var dependenciaId: Int? = null,
    var salida: String? = null,
    var obraSocial: Long? = null,
    var actividadAfip: Int? = null,
    var localidadAfip: Int? = null,
    var situacionAfip: Int = 0,
    var modeloContratacionAfip: Int? = null,
    var directivoEtec: Byte = 0,
    var dependencia: DependenciaDto? = null,
    var afipSituacion: AfipSituacionDto? = null

) {
    fun getApellidoNombre(): String {
        return MessageFormat.format("{0}, {1}", this.apellido, this.nombre)
    }
}
